python/arcticdb/storage_fixtures/api.py

```diff
@@ -0,0 +1,182 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file LICENSE.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+
+import functools
+from typing import Dict, Pattern, Callable, List
+from contextlib import AbstractContextManager, contextmanager
+from abc import abstractmethod
+from enum import Enum
+
+from .utils import handle_cleanup_exception
+from arcticdb import Arctic
+from arcticdb.config import Defaults
+from arcticdb.version_store import NativeVersionStore
+from arcticdb.version_store.helper import ArcticMemoryConfig
+from arcticdb.util.test import apply_lib_cfg
+from arcticc.pb2.storage_pb2 import EnvironmentConfigsMap
+
+
+
+class ArcticUriFields(Enum):
+    """For use with ``replace_uri_field()`` below"""
+
+    HOST = "HOST"
+    USER = "USER"
+    PASSWORD = "PASSWORD"
+    BUCKET = "BUCKET"
+    CA_PATH = "CA_PATH"
+
+    def __str__(self):
+        return self.value
+
+
+class _SaferContextManager(AbstractContextManager):
+    def __enter__(self):
+        try:
+            self._safe_enter()
+            return self
+        except Exception as e:
+            self.__exit__(type(e), e, None)
```

Comment:
Why do we need this exception raising logic? The exception will be raised even without this as long as `__exit__` returns anything other than `True`.